manager/allocator/cnmallocator/networkallocator.go

```diff
@@ -426,13 +425,13 @@ func (na *cnmNetworkAllocator) IsServiceAllocated(s *api.Service, flags ...func(
 	// in VIP allocated set then we return to be allocated to make
 	// sure the allocator triggers networkallocator to free up the
 	// resources if any.
-	if s.Spec.Endpoint != nil && s.Spec.Endpoint.Mode == api.ResolutionModeDNSRoundRobin {
+	if s.Spec.Endpoint.Mode == api.ResolutionModeDNSRoundRobin {
 		if _, ok := na.services[s.ID]; ok {
 			return false
 		}
 	}
 
-	if (s.Spec.Endpoint != nil && len(s.Spec.Endpoint.Ports) != 0) ||
+	if len(s.Spec.Endpoint.Ports) != 0 ||
 		(s.Endpoint != nil && len(s.Endpoint.Ports) != 0) {
```

Comment:
I guess it makes sense to keep the EndpointSpec in Endpoint to be a pointer, but this makes it inconsistent.